plinth/modules/dynamicdns/dynamicdns.py

```diff
@@ -0,0 +1,394 @@
+
+    enabled = forms.BooleanField(label=_('Enable Dynamic DNS'),
+                                 required=False)
+
+    service_type = forms.ChoiceField(label=_('Service type'),
+                                     help_text=_(hlp_services),
+                                     choices=provider_choices)
+
+    dynamicdns_server = TrimmedCharField(
+        label=_('GnudIP Server Address'),
+        required=False,
+        help_text=_(hlp_server),
+        validators=[
+            validators.RegexValidator(r'^[\w-]{1,63}(\.[\w-]{1,63})*$',
+                                      _('Invalid server name'))])
+
+    dynamicdns_update_url = TrimmedCharField(label=_('Update URL'),
+                                             required=False,
+                                             help_text=_(hlp_updt_url))
+
+    disable_SSL_cert_check = forms.BooleanField(label=_('accept all SSL \
+                                                        certificates'),
+                                                help_text=_(hlp_disable_ssl),
+                                                required=False)
+
+    use_http_basic_auth = forms.BooleanField(label=_('use HTTP basic \
+                                                      authentication'),
+                                             help_text=_(hlp_http_auth),
+                                             required=False)
+
+    dynamicdns_domain = TrimmedCharField(
+        label=_('Domain Name'),
+        help_text=_(hlp_domain),
+        required=False,
+        validators=[
+            validators.RegexValidator(r'^[\w-]{1,63}(\.[\w-]{1,63})*$',
+                                      _('Invalid domain name'))])
+
+    dynamicdns_user = TrimmedCharField(
+        label=_('Username'),
+        required=False,
+        help_text=_(hlp_user))
+
+    dynamicdns_secret = TrimmedCharField(
+        label=_('Password'), widget=forms.PasswordInput(),
+        required=False,
+        help_text=_(hlp_secret))
+
+    showpw = forms.BooleanField(label=_('show password'),
+                                required=False)
+
+    dynamicdns_ipurl = TrimmedCharField(
+        label=_('IP check URL'),
+        required=False,
+        help_text=_(hlp_ipurl),
+        validators=[
+            validators.URLValidator(schemes=['http', 'https', 'ftp'])])
+
+    def clean(self):
+        cleaned_data = super(ConfigureForm, self).clean()
+        dynamicdns_secret = cleaned_data.get('dynamicdns_secret')
+        dynamicdns_update_url = cleaned_data.get('dynamicdns_update_url')
+        dynamicdns_user = cleaned_data.get('dynamicdns_user')
+        dynamicdns_domain = cleaned_data.get('dynamicdns_domain')
+        dynamicdns_server = cleaned_data.get('dynamicdns_server')
+        service_type = cleaned_data.get('service_type')
+        old_dynamicdns_secret = self.initial['dynamicdns_secret']
+
+        """clear the fields which are not in use"""
+        if service_type == '1':
```

Comment:
The code would be easier understandable if you would use abbreviations instead of numbers for the select field (like 'gnudip', 'noip', ..)